src/types/k8s.go

```diff
@@ -23,18 +49,30 @@ type ZarfState struct {
 // DeployedPackage contains information about a Zarf Package that has been deployed to a cluster
 // This object is saved as the data of a k8s secret within the 'Zarf' namespace (not as part of the ZarfState secret).
 type DeployedPackage struct {
-	Name       string      `json:"name"`
-	Data       ZarfPackage `json:"data"`
-	CLIVersion string      `json:"cliVersion"`
-
-	DeployedComponents []DeployedComponent `json:"deployedComponents"`
-	ConnectStrings     ConnectStrings      `json:"connectStrings,omitempty"`
+	Name               string                        `json:"name"`
+	Data               ZarfPackage                   `json:"data"`
+	CLIVersion         string                        `json:"cliVersion"`
+	Generation         int                           `json:"generation"`
+	DeployedComponents []DeployedComponent           `json:"deployedComponents"`
+	ComponentWebhooks  map[string]map[string]Webhook `json:"componentWebhooks,omitempty"`
+	ConnectStrings     ConnectStrings                `json:"connectStrings,omitempty"`
 }
 
 // DeployedComponent contains information about a Zarf Package Component that has been deployed to a cluster.
 type DeployedComponent struct {
-	Name            string           `json:"name"`
-	InstalledCharts []InstalledChart `json:"installedCharts"`
+	Name               string           `json:"name"`
+	InstalledCharts    []InstalledChart `json:"installedCharts"`
+	Images             []string         `json:"images"`
```

Comment:
For the ECR external registry use case (and possibly other external registry hooks), it is helpful to have the images data as part of the `DeployedComponents` struct because that is the primary data structure being used to have Zarf and Pepr communicate with one another and tell the other when it can operate.

In the ECR case, we only want the webhook to act upon `Deploying` components that have images. Fetching this from the package data is not as reliable because all components (and any of their images) in a package are stored in the package secret, whether they are actually being deployed or not.

Definitely open to different/better ways to handle this, but I think this approach will allow us to keep our webhook logic more simple

Reply:
It does take more work to lookup the Images for a component but given we know the component name isn't that bad - having Images stand alone feels weird because it is just one part of that component (what if a Pepr webhook needed `repos` instead?) - IMO Images should either be removed or the entire component definition should be inside of the `DeployedComponents` struct (which might be useful for rectifying failed deployments as well).

Reply:
Sounds good. Removed the Images field from the DeployedComponent struct